dev/TabView/TabViewItem.cpp

```diff
@@ -84,6 +92,15 @@ void TabViewItem::OnApplyTemplate()
 
 void TabViewItem::OnIsSelectedPropertyChanged(const winrt::DependencyObject& sender, const winrt::DependencyProperty& args)
 {
+    if (IsSelected())
+    {
+        SetValue(winrt::Canvas::ZIndexProperty(),box_value(20));
```

Comment:
can this be done in the visual state ?

Reply:
If Tab Placement for the bottom or sides is added, there will need to be new visual states added.  Would that cause any issues?

Reply:
> can this be done in the visual state ?

If we want the inset corners to be able to register hits, so that clicking on them does not switch the selection to the item they are on top, unfortunately not.

> If Tab Placement for the bottom or sides is added, there will need to be new visual states added. Would that cause any issues?

Since we are talking about ZIndex here, this won't be problematic with the bottoms or side. The inset corners may become problematic, but that is something we need to deal with later.

Reply:
> > can this be done in the visual state ?
> 
> If we want the inset corners to be able to register hits, so that clicking on them does not switch the selection to the item they are on top, unfortunately not.

I don't understand this, I agree the property needs to be set to achieve this but it should be settable from the visual state shouldn't it?

Reply:
Setting the ZIndex does not work in the visual state as we can't set the ZIndex of the TabViewItem itsself. Setting it on the LayoutRoot does not work, as setting ZIndex of the layoutroot only "lifts" the LayoutRoot relative to other childs with same parent. Since the TabViewItem needs to be lifted relative to the other TabViewItems, we need to set that items ZIndex.

Reply:
Sorry I didn't see this a while ago.. I thought adding this to the selected state in the TabViewItem's VSM would set the ZIndex on the TVI itself:

`<Setter Property="Canvas.ZIndex" Value="20"/>`

I think ommiting a target makes the property get set on the TVI itself

Reply:
Trying to define it using `Setter.Property` results in the following compile error:

1>D:\Projects\microsoft-ui-xaml\BuildOutput\Debug\\x64\Microsoft.UI.Xaml\rs2_generic.xaml(556,54): XamlCompiler error WMC0610: The XAML Binary Format (XBF) generator reported syntax error '0x09ff'

Any ideas @StephenLPeters what could cause this?

Reply:
I suspect the syntax is a bit different but I'm not sure what it is precisely, I think its fine to leave as is.